hbase-common/src/main/java/org/apache/hadoop/hbase/HConstants.java

```diff
@@ -1049,6 +1049,16 @@
 
   public static final float HFILE_BLOCK_CACHE_SIZE_DEFAULT = 0.4f;
 
+  /**
+   * Configuration key for setting the fix size of the block size, default do nothing and it should
+   * be explicitly set by user or only used within ClientSideRegionScanner. if it's set less than
+   * current max on heap size, it overrides the max size of block cache
+   */
+  public static final String HBASE_BLOCK_CACHE_FIXED_SIZE_KEY =
```

Comment:
We normally avoid adding these constants to Public HConstants,

Reply:
@anoopsjohn let me know if you have further comments on this size parameter, the latest change has the same limit as same as the `hfile.block.cache.size` that it won't go beyond the max available heap